pkg/detectors/yandex/yandex.go

```diff
@@ -63,6 +57,34 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return results, nil
 }
 
+func verifyMatch(ctx context.Context, client *http.Client, apiKey string) (bool, error) {
+	// Reference: https://yandex.com/dev/dictionary/doc/dg/reference/getLangs.html
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("https://dictionary.yandex.net/api/v1/dicservice.json/getLangs?key=%s", apiKey), http.NoBody)
+	if err != nil {
+		return false, err
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		return true, nil
+	case http.StatusPaymentRequired:
+		return true, fmt.Errorf("blocked due to billing issue/quota exceeded: %d", res.StatusCode)
```

Comment:
If we're certain that the API returns this error when the key is valid but blocked due to billing issues or quota limits, then I don't think we should surface it as an error to the user. It's not a failure to verify the key itself.